presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java

```diff
@@ -103,6 +103,7 @@ public static DistributedQueryRunner createQueryRunner(Iterable<TpchTable<?>> ta
                     .put("hive.time-zone", TIME_ZONE.getID())
                     .put("hive.security", security)
                     .put("hive.max-partitions-per-scan", "1000")
+                    .put("hive.assume-canonical-partition-keys", "true")
```

Comment:
It would be nice to verify that it actually happen. E.g. create two partitions and then somehow test that only one was read from Hive. Not sure it is possible this or some other way but without it is hard to be sure we have coverage.